### PR TITLE
Execute project dependency command from output directory for Desktop.

### PR DESCRIPTION
**Pull request by Sridhar-MS**
Created a new command resolver called OutputPathCommandResolver which
reolves a command from the project build output directory.

Made changes ProjectDependenciesCommandFactory to use
OutputPathCommandResolver if the target framework is Desktop.

Fixes - #2126

Still need to add tests.

cc @piotrpMSFT @brthor @bricelam

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2297)

### CODE REVIEW

**Comment by brthor** [comment]
What do you think of adding one test to dotnet.Tests for the desktop scenario that uses this? Would give confidence here I think.

**Comment by Sridhar-MS** [comment]
@brthor checkout the existing tests I have modified to test the dependency tool load location.

File: src/Microsoft.DotNet.Cli.Utils/CommandResolution/OutputPathCommandResolver.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class OutputPathCommandResolver : AbstractPathBasedCommandResolver
```
Comment (brthor) [inline src/Microsoft.DotNet.Cli.Utils/CommandResolution/OutputPathCommandResolver.cs]: :+1:

**Comment by brthor** [comment]
LGTM :shipit:

**Comment by brthor** [comment]
:shipit:

File: src/Microsoft.DotNet.Cli.Utils/ProjectDependenciesCommandFactory.cs [hunk 2]
```diff
@@ -87,12 +88,29 @@ public class ProjectDependenciesCommandFactory : ICommandFactory
...
-        private ICommandResolver GetProjectDependenciesCommandResolver()
+        private ICommandResolver GetProjectDependenciesCommandResolver(NuGetFramework framework)
         {
             var environment = new EnvironmentProvider();
-            var packagedCommandSpecFactory = new PackagedCommandSpecFactory();
 
-            return new ProjectDependenciesCommandResolver(environment, packagedCommandSpecFactory);
+            if (framework.IsDesktop())
+            {
+                var platformCommandSpecFactory = default(IPlatformCommandSpecFactory);
```
Comment (davidfowl) [inline src/Microsoft.DotNet.Cli.Utils/ProjectDependenciesCommandFactory.cs]: nit: 

`IPlatformCommandSpecFactory platformCommandSpecFactory = null` 

😄

Comment (Sridhar-MS) [inline src/Microsoft.DotNet.Cli.Utils/ProjectDependenciesCommandFactory.cs]: Fixed the nit

**Comment by Sridhar-MS** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by Sridhar-MS** [comment]
@dotnet-bot test Windows_NT x64 Release Build please

**Comment by Sridhar-MS** [comment]
@dotnet-bot test  Windows_NT x86 Debug Build please

**Pull request merged**